packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -66,20 +74,34 @@ export class OpenID4VCIClient {
     credentialIssuer?: string;
     pkce?: PKCEOpts;
     authorizationRequest?: AuthorizationRequestOpts; // Can be provided here, or when manually calling createAuthorizationUrl
+    jwk?: JWK;
+    endpointMetadata?: EndpointMetadataResult;
+    accessTokenResponse?: AccessTokenResponse;
+    authorizationRequestOpts?: AuthorizationRequestOpts;
+    authorizationURL?: string;
   }) {
-    this._credentialOffer = credentialOffer;
     const issuer = credentialIssuer ?? (credentialOffer ? getIssuerFromCredentialOfferPayload(credentialOffer.credential_offer) : undefined);
     if (!issuer) {
       throw Error('No credential issuer supplied or deduced from offer');
     }
-    this._credentialIssuer = issuer;
-    this._kid = kid;
-    this._alg = alg;
-    // TODO: We need to refactor this and always explicitly call createAuthorizationRequestUrl, so we can have a credential selection first and use the kid as a default for the client id
-    this._clientId = clientId ?? (credentialOffer && getClientIdFromCredentialOfferPayload(credentialOffer.credential_offer)) ?? kid?.split('#')[0];
-    this._pkce = { ...this._pkce, ...pkce };
-    this._authorizationRequestOpts = this.syncAuthorizationRequestOpts(authorizationRequest);
-    debug(`Authorization req options: ${JSON.stringify(this._authorizationRequestOpts, null, 2)}`);
+    this._state = {
+      credentialOffer,
+      credentialIssuer: issuer,
+      kid,
+      alg,
+      // TODO: We need to refactor this and always explicitly call createAuthorizationRequestUrl, so we can have a credential selection first and use the kid as a default for the client id
+      clientId: clientId ?? (credentialOffer && getClientIdFromCredentialOfferPayload(credentialOffer.credential_offer)) ?? kid?.split('#')[0],
+      pkce: { disabled: false, codeChallengeMethod: CodeChallengeMethod.S256, ...pkce },
+      jwk,
+      endpointMetadata,
+      accessTokenResponse,
+      authorizationURL
+    }
+    this._state = {
+      ...this._state,
+      authorizationRequestOpts: authorizationRequestOpts ?? this.syncAuthorizationRequestOpts(authorizationRequest),
```

Comment:
Why not assign the Opts to this._state directly. We are still in the constructor

Reply:
made some changes, let me know if this is ok